### PR TITLE
Implement basic map view

### PR DESCRIPTION
**Pull request by hayata-suenaga**
### Details

This PR creates a cross-platform `MapView` component.

### Related Issues

https://github.com/Expensify/App/issues/22703

### Manual Tests

The component is tested in [this App PR](https://github.com/Expensify/App/pull/24306). I can open a follow up PR to include automated tests on the library itself after high priority issues are done. However, I don't have a good idea on how to test visual elements on Map.

### Linked PRs

https://github.com/Expensify/App/pull/24306

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ module.exports = {
             },
         ],
         'no-invalid-this': 'error',
+        'no-use-before-define': 'off',
```
Comment (thienlnam) [inline .eslintrc.js]: Are these the same ones in App?

File: src/MapView/consts.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import {MapViewProps} from './MapViewTypes';
```
Comment (thienlnam) [inline src/MapView/consts.ts]: To be consistent, lets rename this file to CONST?

File: src/MapView/utils.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// ESLint rule forces the usage of default export if there is only one export.
+// But not using default export here as more util functions might be added later.
+// eslint-disable-next-line import/prefer-default-export
+export const getBounds = (waypoints: Array<[number, number]>): {southWest: [number, number]; northEast: [number, number]} => {
```
Comment (thienlnam) [inline src/MapView/utils.ts]: Let’s try to keep the style consistent as the App repo - so making this a function

and then export default {
Function
}

**Review by thienlnam — CHANGES_REQUESTED** [review]
Looks good, can’t speak much to the Mapbox stuff but left some comments regarding keeping conventions same as App styling

File: .eslintrc.js (hunk 1 above)
Comment (hayata-suenaga) [reply .eslintrc.js]: nop 😅 turning it back one

**Review by thienlnam — APPROVED** [review]

File: src/MapView/MapView.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        () => ({
+            flyTo: (location: [number, number], animationDuration?: number) => cameraRef.current?.flyTo(location, animationDuration),
+        }),
+        [],
+    );
+
+    // Initialize Mapbox on first mount
+    useEffect(() => {
+        Mapbox.setAccessToken(accessToken);
+    }, []);
```
Comment (allroundexperts) [inline src/MapView/MapView.tsx:47]: This code caused https://github.com/Expensify/App/issues/25732. Map box API doesn't seem to support `flyTo` well enough on native platforms.

Comment (hayata-suenaga) [reply src/MapView/MapView.tsx:47]: it's rather the issue with the libraries themself than a regression but yea we should have tested this thoroughly

**Pull request merged**